peekingduck/training/src/data/data_adapter.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import Any, Optional
+from omegaconf import DictConfig
+from torch.utils.data import DataLoader
+from albumentations import Compose
+
+from src.data.base import AbstractDataSet
+from src.data.dataset import TFImageClassificationDataset
+
+
+@dataclass
+class DataAdapter:
+    """Adapter for different framework"""
+
+    cfg: DictConfig = field()
```

Comment:
Is there a reason to use `field()` instead of just type hinting as `cfg: DictConfig`?

I think some usages of `field()` is when your attribute is mutable, setting default values and controlling serialization in `__repr__`.